### ISSUE TITLE
Cannot use custom Eth RPC Endpoint over HTTPS with Certificated trusted by system Windows Install

### ISSUE THREAD

**Issue by MysticRyuujin** [rotki/rotki#1223]
## Problem Definition
I'm not sure if this is related to #889 or it's something else? But...I have my own private endpoints on my network secured via a local Root CA. My system (Windows) trusts my root CA certificate and all certificates signed by it. But, it doesn't look like Rotki supports this?

Using http works fine but https results in:
`Setting not saved: Error setting rpc endpoint Failed to connect to ethereum node at endpoint https://<node>`

I assume that the issue is that Rotki doesn't use the system certificate store on Windows? If that's the case maybe that's more of a feature request than a bug?

I don't consider it a high priority or anything, it's local traffic, just thought I would report it.

### System Description

Here add a detailed description of your system, e.g. output of the following script:

```
rotkehlchen-1.6.1-windows.exe 
```

**Comment by LefterisJP** [comment]
Hey @MysticRyuujin nice to see you here!

Would it be possible to include the rotkehlchen.log of a run where you get this error? I would like to see what is the exact reason of the failure. Hopefully something is logged there.

To get the log location you can check the top menu. "Help" -> "Logs Location"

**Comment by MysticRyuujin** [comment]
Hmmm... all I see is this:
```
26/07/2020 14:31:23 Central Daylight Time -- INFO:rotkehlchen.api.server.pywsgi:127.0.0.1 - - [2020-07-26 14:31:23] "OPTIONS /api/1/settings HTTP/1.1" 200 319 0.000500
26/07/2020 14:31:23 Central Daylight Time -- DEBUG:web3.providers.HTTPProvider:Making request HTTP. URI: https://<node>, Method: web3_clientVersion
26/07/2020 14:31:23 Central Daylight Time -- WARNING:rotkehlchen.chain.ethereum.manager:Could not connect to an ethereum node. Will use etherscan only
26/07/2020 14:31:23 Central Daylight Time -- DEBUG:rotkehlchen.api.rest:Request successful response={"result": null, "message": "Failed to connect to ethereum node at endpoint https://<node>"}, status_code=409
```

Here's what it looks live if I use HTTP:
```
26/07/2020 14:42:39 Central Daylight Time -- DEBUG:web3.providers.HTTPProvider:Making request HTTP. URI: http://<node>, Method: web3_clientVersion
26/07/2020 14:42:39 Central Daylight Time -- DEBUG:web3.providers.HTTPProvider:Getting response HTTP. URI: http://<node>, Method: web3_clientVersion, Response: {'jsonrpc': '2.0', 'result': 'Nethermind/v1.8.71-0-c41e344d2-20200723/X64-Linux/Core3.1.5', 'id': 0}
```

Really doesn't give any good info as to why.

Proof the endpoint works over HTTPS (notice I also do not include `-k` or `--insecure`):

```
curl -H "Content-Type: application/json" -d '{"id": 1, "method": "web3_clientVersion", "params": []}' https://<node>
{"jsonrpc":"2.0","result":"Nethermind/v1.8.71-0-c41e344d2-20200723/X64-Linux/Core3.1.5","id":1}
```

**Comment by MysticRyuujin** [comment]
I did a wireshark against the traffic, this is what I see:
![image](https://user-images.githubusercontent.com/8877131/88487936-39f5d680-cf4f-11ea-8925-3c98757d4360.png)

**Comment by LefterisJP** [comment]
All right I see. From your description, and the wireshark message and from the chat we had in Twitter it seems as if the python requests library does not recognize the local Windows certificate store.

And we do fail when the SSL Certificate can't be verified.

For your particular case [this](https://pypi.org/project/python-certifi-win32/) python package may help as it will patch python requests to accept the local windows certificate store.

We should include it in the windows build only.

So the task here is to:

- [ ] Include `python-certifi-win32` only for the windows builds.
- [ ] Create a test binary for you to test
- [ ] Provide better error message when there is a failure to connect to the RPC endpoint. At least log something useful to the logs.

**Comment by LefterisJP** [comment]
Hey @MysticRyuujin when you find some time can you please check this experimental dev binary for Windows? https://we.tl/t-bzORv2yTKg

Please use it with a fresh account, as if you use it with a v1.6.1 account it will be upgraded to a new DB version and then you won't be able to go back to the v1.6.1 binary (unless we do some hacks).

Tell me if by using this binary the `https://node` version works.

**Comment by MysticRyuujin** [comment]
Well...I was having a bunch of issues creating a new profile and/or logging into a new profile...but I eventually got a new profile created....

Those issues aside, no, it did not fix the issue. I get the same result and the same Wireshark output (Unknown CA)

**Comment by MysticRyuujin** [comment]
Maybe an alternative approach is to simply allow the JSON RPC endpoint to ignore certs? Unless you've got any other ideas haha

**Comment by LefterisJP** [comment]
> I get the same result and the same Wireshark output (Unknown CA)

Hmm that's unfortunate. From the description of the package, it should have solved your issue. Sucks I can't actually reproduce this in any way. We could ignore SSL Certificate errors as an option.

> Well...I was having a bunch of issues creating a new profile and/or logging into a new profile...but I eventually got a new profile created....

Can you tell me what those issues were? This is basically a snapshot of develop branch and has some changes so I am wondering what it is you saw.

**Comment by LefterisJP** [comment]
> Can you tell me what those issues were? This is basically a snapshot of develop branch and has some changes so I am wondering what it is you saw.

We had a call and I figured out the problem. We were doing quite a few synchronous calls during new user creation/unlock. This has been fixed in develop and will be released in v1.6.2.

----

As for the original issue it's pushed for later as I am not currently sure what's the best way to handle this.

**Comment by LefterisJP** [comment]
I am closing this issue since there seems to not have been a way to accomplish this in Windows and nobody else has complained about this in the 3+ years since.
